### ISSUE TITLE
False positive MissingTemplateParam

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1238]
https://getpsalm.org/r/a03973a98b
```php
<?php
  /**
   * @template TKey
   * @template TValue
   */
  interface Selectable {}

  /**
   * @template T
   * @template-implements Selectable<int,T>
   */
  class Repository implements Selectable {}

  interface SomeEntity {}

  /**
   * @template-extends Repository<SomeEntity>
   */
  class SomeRepository extends Repository {}
```

**Expected**: no issues, as `SomeRepository` is `Selectable<int,SomeEntity>`
**Actual**: 
```
Psalm output (using commit cd0a744): 

ERROR: MissingTemplateParam - 19:9 - Selectable has unextended template params, expecting 2
```
